features/netsocket/TLSSocket.h

```diff
@@ -75,6 +75,9 @@ class TLSSocket : public TLSSocketWrapper {
      *  Initiates a connection to a remote server specified by either
      *  a domain name or an IP address and a port.
      *
+     *  @note: In case connect() returns an NSAPI_ERROR_AUTH_FAILURE,
+     *  a new socket should be created before reconnecting.
```

Comment:
Should this sentence reflect more about this socket, instead of how to proceed?

For example: "In case connect() returns error, you must stop using the socket and free it either by calling its close() function or destroying it"

Reply:
Shouldn't we stick to  NSAPI_ERROR_AUTH_FAILURE? Other errors do not really render the socket useless. It is just a recommendation in Socket.h to create a new one... 
Otherwise, I agree. Is it okay to be a little more formal, like so: "In case connect() returns NSAPI_ERROR_AUTH_FAILURE, the socket must be freed either by calling close() or destroying it"?

Reply:
Yes. That makes sense.